### DISCUSSION TITLE
How are ./. (missing) genotypes in families handled?

### DISCUSSION THREAD

**Discussion by jxchong** [broadinstitute/seqr#2505] [Q&A] [answered]
How are missing genotypes/alleles (./.) handled? The interface seems to be indicating that it requires all individuals in the family to have called genotypes to successfully search for a variant but wasn't sure if there is additional logic on the backend. 

Examples for a trio:
If mom is ./. dad is ./. and affected child is 0/1, does the variant appear when you do a variant search using the the de novo inheritance model?
If mom is ./. dad is 0/0 and affected child is 0/1, does the variant appear when you do a variant search using the the de novo inheritance model?
If mom is ./. affected dad is 0/1 and affected child is 0/1, does the variant appear in the autosomal dominant variant search?

**Comment by hanars** [comment, marked as answer]
Variants with missing genotypes are not returned in inheritance based queries. None of the scenarios you listed would return the variant (although scenario 3 "dad is 0/1 and affected child is 0/1" will never return in dominant search, regardless of mom's genotype)

The relevant backend code is here: https://github.com/broadinstitute/seqr/blob/master/seqr/utils/elasticsearch/constants.py#L20

**Reply by jxchong** [reply]
Ok thanks! Bad typing for scenario 3 - it's supposed to be: 
unaffected mom ./.
affected dad 0/1
affected child 0/1

Is this on a per-family basis? So if another, unrelated individual (Family 2) in the VCF is ./. for the same variant, but the family's genotypes are all present (mom 0/0, dad 0/0, child 0/1), would that variant be pulled out by a de novo search?

**Reply by hanars** [reply]
Variants are only returned in families you explicitly search for (which can include searching for all families in a project). If you are searching in multiple families, variants will be returned if they match any family, but genotype and tag information will only be shown for the families that match. So in that scenario the variant would be returned by any de novo search that includes family 1 but not a search that only includes family 2. If you did a search that includes both family 1 and family 2, the variant would be returned showing only the genotype information for family 1
